### ISSUE TITLE
Failed to select decimal value with hive when scale in filter is different with hive DDL schema

### ISSUE THREAD

**Issue by qphien** [apache/iceberg#1699]
Related to issue: https://github.com/apache/iceberg/pull/1326
```
hive (iceberg)> desc decimal_table;
OK
col_name	data_type	comment
val             decimal(7,2)    from deserializer

hive (iceberg)> select * from decimal_table where val > 100.1;
OK
decimal_table.val
Failed with exception java.io.IOException:org.apache.iceberg.exceptions.ValidationException: Invalid value for conversion to type decimal(7, 2): 100.1 (java.math.BigDecimal)
```

With 100.1 cast as decimal(7, 2), hive still throws the same exception.

```
hive (iceberg)> select * from decimal_table where val > cast(100.1 as decimal(7,2));
OK
decimal_table.val
Failed with exception java.io.IOException:org.apache.iceberg.exceptions.ValidationException: Invalid value for conversion to type decimal(7, 2): 100.1 (java.math.BigDecimal)
```

**Comment by marton-bod** [comment]
Thanks for flagging this issue @qphien . I've started investigating it.

**Comment by qphien** [comment]
This issuse has been fixed by #1742. Close this issue
